packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts

```diff
@@ -460,4 +462,100 @@ class DataStorageAuthWithTriggerTestProject extends TestProjectBase {
       throw err;
     }
   };
+
+  private createSQSQueue = async () => {
+    const createQueueResponse = await this.sqsClient.send(
+      new CreateQueueCommand({ QueueName: this.queueName })
+    );
+    this.queueUrl = createQueueResponse.QueueUrl ?? '';
+  };
+
+  private deleteSQSQueue = async () => {
+    if (this.queueUrl) {
+      await this.sqsClient.send(
+        new DeleteQueueCommand({
+          QueueUrl: this.queueUrl,
+        })
+      );
+    }
+  };
+
+  private addLambdaPermissionForSQS = async (backendId: BackendIdentifier) => {
+    this.funcWithSchedulePhysicalIds =
+      await this.resourceFinder.findByBackendIdentifier(
+        backendId,
+        'AWS::Lambda::Function',
+        (name) => name.includes('funcWithSchedule')
+      );
+    const getQueueAttributesResponse = await this.getSQSQueueAttributes(
+      this.queueUrl
+    );
+    const queueArn = getQueueAttributesResponse.Attributes?.QueueArn ?? '';
+
+    const sqsPolicy = new PolicyDocument({
+      statements: [
+        new PolicyStatement({
+          actions: ['sqs:SendMessage'],
+          effect: Effect.ALLOW,
+          resources: [queueArn],
+        }),
+      ],
+    });
+
+    const getFunctionConfigurationResponse = await this.lambdaClient.send(
+      new GetFunctionConfigurationCommand({
+        FunctionName: this.funcWithSchedulePhysicalIds[0],
+      })
+    );
+    const lambdaRole = getFunctionConfigurationResponse.Role
+      ? getFunctionConfigurationResponse.Role.split('/')[1]
+      : '';
```

Comment:
Can't we get the `role` using `findByBackendIdentifier`? The role should be part of the stack.

Reply:
Moving SQS and granting the schedule function `SendMessage` permission to the backend app eliminated this